src/semantic_version.cr

```diff
@@ -19,7 +23,18 @@ struct SemanticVersion
   # The pre-release version of this semantic version
   getter prerelease : Prerelease
 
-  # Parses a `SemanticVersion` from the given semantic version string
+  # Checks if *str* is a valid semantic version.
+  #
+  # ```
+  # require "semantic_version"
+  #
+  # SemanticVersion.valid?("1.15.0") # => true
+  # SemanticVersion.valid?("1.2")    # => false
+  def self.valid?(str : String) : Bool
```

Comment:
Do we need to add the method `valid?` if `parse?` can be used in exactly the same way? And I think currently there isn't any significant performance benefit to it either

Reply:
I think it's more semantics, for example, `Hash#has_key?` vs `Hash#[]?`: they can be used in the same way and have the same underlying implementation, but the methods themselves convey different things. Comparing performance feels like splitting hairs, I agree that it's likely minimal.

Reply:
I think this is a valid point. I would read the semantic meaning of `parse?` as "can it parse this?". If not, the return value is falsey. If yes, it's the parse result.

A differentiation into a specific bool method like `valid?` is particularly relevant when validation can be implemented more efficiently than parsing.